pkg/bucket/bandwidth/measurements/object_measurement.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package measurements
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	// BetaObject is the weight used to calculate exponential moving average
+	BetaObject = 0.1 // Number of averages considered = 1/(1-BetaObject)
+)
+
+// ObjectMeasurement captures the details for bandwidth measurement for objects.
+type ObjectMeasurement struct {
```

Comment:
Didn't we agree to take out the individual object tracking? As far as I can tell this doesn't do anything other than burn cpu cycles. ie it isn't used for reporting.

Also, this duplicates all code of `BucketMeasurement`. Combine them into `Measurement` that doesn't care what it is tracking.

Reply:
Bucket measurement is windowed and across all objects. Object measurement is per object. Sum of duration for objects can be greater than wall clock time but not for bucket. There is a difference.
I am waiting to see how mc evolves, it might make sense to be able to list on a per bucket basis which objects are being tracked.